server/chat-plugins/helptickets-auto.ts

```diff
@@ -0,0 +1,670 @@
+					if (punishment[k as keyof AutoPunishment] === curP[k as keyof AutoPunishment]) {
+						matches++;
+					}
+				}
+				if (matches === Object.keys(punishment).length) {
+					return this.errorReply(`That punishment is already added.`);
+				}
+			}
+			settings.punishments.push(punishment as AutoPunishment);
+			saveSettings();
+			this.privateGlobalModAction(
+				`${user.name} added a ${punishment.punishment} punishment to the Artemis helpticket handler.`
+			);
+			this.globalModlog(`AUTOHELPTICKET ADDPUNISHMENT`, null, visualizePunishment(punishment as AutoPunishment));
+		},
+		dp: 'deletepunishment',
+		deletepunishment(target, room, user) {
+			checkAccess(this);
+			const num = parseInt(target) - 1;
+			if (isNaN(num)) return this.parse(`/h autohelpticket`);
+			const punishment = settings.punishments[num];
+			if (!punishment) return this.errorReply(`There is no punishment at index ${num + 1}.`);
+			settings.punishments.splice(num, 1);
+			this.privateGlobalModAction(
+				`${user.name} removed the Artemis helpticket ${punishment.punishment} punishment indexed at ${num + 1}`
+			);
+			this.globalModlog(`AUTOHELPTICKET REMOVE`, null, visualizePunishment(punishment));
+		},
+		vp: 'viewpunishments',
+		viewpunishments() {
+			checkAccess(this);
+			let buf = `<strong>Artemis helpticket punishments</strong><hr />`;
+			if (!settings.punishments.length) {
+				buf += `None.`;
+				return this.sendReplyBox(buf);
+			}
+			buf += settings.punishments.map(
+				(curP, i) => `<strong>${i + 1}:</strong> ${visualizePunishment(curP)}`
+			).join('<br />');
+			return this.sendReplyBox(buf);
+		},
+		togglepunishments(target, room, user) {
+			checkAccess(this);
+			let message;
+			if (this.meansYes(target)) {
+				if (settings.applyPunishments) {
+					return this.errorReply(`Automatic punishments are already enabled.`);
+				}
+				settings.applyPunishments = true;
+				message = `${user.name} enabled automatic punishments for the Artemis ticket handler`;
+			} else if (this.meansNo(target)) {
+				if (!settings.applyPunishments) {
+					return this.errorReply(`Automatic punishments are already disabled.`);
+				}
+				settings.applyPunishments = false;
+				message = `${user.name} disabled automatic punishments for the Artemis ticket handler`;
+			} else {
+				return this.errorReply(`Invalid setting. Must be 'on' or 'off'.`);
+			}
+			this.privateGlobalModAction(message);
+			this.globalModlog(`AUTOHELPTICKET TOGGLE`, null, settings.applyPunishments ? 'on' : 'off');
+			saveSettings();
+		},
+	},
+	autohelptickethelp: [
+		`/aht addpunishment [args] - Adds a punishment with the given [args]. Requires: whitelist &`,
+		`/aht deletepunishment [index] - Deletes the automatic helpticket punishment at [index]. Requires: whitelist &`,
+		`/aht viewpunishments - View automatic helpticket punishments. Requires: whitelist &`,
+		`/aht togglepunishments [on | off] - Turn [on | off] automatic helpticket punishments. Requires: whitelist &`,
```

Comment:
Would it be useful to add `add`, `delete`, `view`, and `toggle` and aliases? That seems at least as intuitive as `vp` to me.

Reply:
Not really. Shorter aliases are preferable when using multiple of these commands.

Reply:
`add` is barely longer than `ap` and makes much more sense; we can have both?

Reply:
OK.

Reply:
I would prefer no alias for toggle, though (i intentionally left one off to make it so you had to type the full thing out to use).

Reply:
OK.